rfcs/2736-system_builder_syntax.md

```diff
@@ -0,0 +1,72 @@
+# Feature Name: (fill me in with a unique ident, `system_builder_syntax`)
+
+## Summary
+
+Rewrite the system creation workflow to remove the excess of `add_startup_system_to_stage`-like methods and instead use the `system.stage().label().startup()` builder-like system.
+
+## Motivation
+
+Mostly described in [#1978](https://github.com/bevyengine/bevy/issues/1978).
+
+- Allows us to unify the existing `.label()`, `.after()`, etc. methods and the stage positioning of systems.
+- App would only have 3 methods to add systems: `add_system`, `add_exclusive` and `add_system_set`.
+- Arguably more intuitive design (somewhat backed up by those who commented on #1978).
+
+## User-facing explanation
+
+- `add_startup_system(system)` becomes `add_system(system.startup())`
+- `add_system_to_stage(stage, system)` becomes `add_system(system.stage(stage))`
+- `add_startup_system_to_stage(stage, system)` becomes `add_system(system.startup().stage(stage))`
+- `add_system(system.exclusive_system())` becomes `add_exclusive(system.exclusive_system())`
+- `add_startup_system(system.exclusive_system())` becomes `add_exclusive(system.exclusive_system().startup())`
+- `add_startup_system(system.label(x).before(y))` becomes `add_system(system.startup().label(x).before(y))`
+
+The rest should follow.
```

Comment:
I'd like to see a demonstration of how this works with system sets, and another with states as well.

Reply:
Please see the updated examples :)

Reply:
Awesome, looks good!